pkg/tsdb/prometheus/models/query.go

```diff
@@ -45,16 +45,12 @@ const (
 var safeResolution = 11000
 
 type QueryModel struct {
-	Expr           string `json:"expr"`
-	LegendFormat   string `json:"legendFormat"`
-	Interval       string `json:"interval"`
-	IntervalMS     int64  `json:"intervalMS"`
-	StepMode       string `json:"stepMode"`
-	RangeQuery     bool   `json:"range"`
-	InstantQuery   bool   `json:"instant"`
-	ExemplarQuery  bool   `json:"exemplar"`
-	IntervalFactor int64  `json:"intervalFactor"`
-	UtcOffsetSec   int64  `json:"utcOffsetSec"`
+	dataquery.PrometheusDataQuery
+	LegendFormat *string `json:"legendFormat,omitempty"`
+	RequestId    *string `json:"requestId,omitempty"`
+	// Timezone offset to align start & end time on backend
+	UtcOffsetSec   *int64 `json:"utcOffsetSec,omitempty"`
+	ValueWithRefId *bool  `json:"valueWithRefId,omitempty"`
```

Comment:
```suggestion
	LegendFormat string `json:"legendFormat,omitempty"`
	RequestId    string `json:"requestId,omitempty"`
	// Timezone offset to align start & end time on backend
	UtcOffsetSec   int64 `json:"utcOffsetSec,omitempty"`
	ValueWithRefId bool  `json:"valueWithRefId,omitempty"`
```

It's probably fine to leave these as non-pointers, same as they were before. Codegen produces pointers to represent field optionality (see grafana/grok#1) in the schema. We may not even preserve that behavior long term because of how annoying it is to work with pointers to certain primitive types in Go.